arch/Kconfig

```diff
@@ -62,6 +62,7 @@ config X86
 	select ARCH_HAS_GDBSTUB if !X86_64
 	select ARCH_HAS_TIMING_FUNCTIONS
 	select ARCH_HAS_THREAD_LOCAL_STORAGE
+	select TOOLCHAIN_SUPPORTS_SOFT_FLOAT if !X86_64 && "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
```

Comment:
if the toolchain itself can state that it supports softfloat, there should not be a need for testing against `zephyr` variant here.